cli/azd/pkg/apphost/generate.go

```diff
@@ -274,9 +275,50 @@ func newInfraGenerator() *infraGenerator {
 	}
 }
 
+// pullParentSettings check a special case where:
+// - resource has a parent with type `container.v0`
+// When this is true, the resource should inherit the parent settings (type, bindings, env, inputs, etc.)
+// For example, a `postgres.database.v0` resource that has a parent of type `container.v0` should inherit the
+// container settings from the parent and become a `container.v0` resource.
+func pullParentSettings(resourceName string, resource *Resource, manifest *Manifest) *Resource {
+	if resource.Parent == nil {
+		return resource
+	}
+
+	if *resource.Parent == "" {
+		return resource
+	}
+
+	parentResource := manifest.Resources[*resource.Parent]
+
+	if parentResource.Type != "container.v0" {
+		return resource
+	}
+
+	reNameParentToChild := func(source, parent, child string) string {
+		return strings.ReplaceAll(
+			source, fmt.Sprintf("%s.inputs.", parent), fmt.Sprintf("%s.inputs.", child))
+	}
+
+	reMapEnv := make(map[string]string, len(parentResource.Env))
+	for k, v := range parentResource.Env {
+		reMapEnv[k] = reNameParentToChild(v, *resource.Parent, resourceName)
+	}
+
+	return &Resource{
+		Type:             parentResource.Type,
```

Comment:
For these pointer values, I wonder if we should be cloning the value instead of putting in the same pointer and creating an alias.  It feels like we might want the clone because later we end up rewriting the bindings map as well as individual binding objects.